nixos/modules/services/desktops/pipewire/wireplumber.nix

```diff
@@ -23,51 +23,102 @@ in
         defaultText = lib.literalExpression "pkgs.wireplumber";
         description = lib.mdDoc "The wireplumber derivation to use.";
       };
+
+      configPackages = lib.mkOption {
+        type = lib.types.listOf lib.types.package;
+        default = [ ];
+        description = lib.mdDoc ''
+          List of packages that provide wireplumber configuration, in the form of
+          `share/wireplumber/*/*.lua` files.
+        '';
+      };
+
+      extraLv2Packages = lib.mkOption {
+        type = lib.types.listOf lib.types.package;
+        default = [];
+        example = lib.literalExpression "[ pkgs.lsp-plugins ]";
+        description = lib.mdDoc ''
+          List of packages that provide LV2 plugins in `lib/lv2` that should
+          be made available to wireplumber for [filter chains][wiki-filter-chain].
+
+          Config packages have their required lv2 plugins added automatically,
+          so they don't need to be specified here.
+
+          [wiki-filter-chain]: https://docs.pipewire.org/page_module_filter_chain.html
+        '';
+      };
     };
   };
 
-  config = lib.mkIf cfg.enable {
-    assertions = [
-      {
-        assertion = !config.hardware.bluetooth.hsphfpd.enable;
-        message = "Using Wireplumber conflicts with hsphfpd, as it provides the same functionality. `hardware.bluetooth.hsphfpd.enable` needs be set to false";
-      }
-    ];
-
-    environment.systemPackages = [ cfg.package ];
+  config =
+    let
+      configPackages = cfg.configPackages
+          ++ lib.optional (!pwUsedForAudio) nixosConfigPkg
+          ++ lib.optionals config.services.pipewire.systemWide [ systemwideConfigPkg systemwideBluetoothConfigPkg ];
 
-    environment.etc."wireplumber/main.lua.d/80-nixos.lua" = lib.mkIf (!pwUsedForAudio) {
-      text = ''
+      nixosConfigPkg = pkgs.writeTextDir "share/wireplumber/main.lua.d/80-nixos.lua" ''
```

Comment:
Not a big fan of the name here. Maybe rename the variable and also the file?

Reply:
I've changed the name of this config file / config package variable to `pw-not-for-audio`/`pwNotForAudio`. Hopefully that is a more descriptive name. Just "nixos" was indeed a somewhat unhelpful name for the file/variable. Thanks for pointing that out.

In case you're not happy with the new name just yet, I'd appreciate some hint at what you want from the name / perhaps a suggestion.